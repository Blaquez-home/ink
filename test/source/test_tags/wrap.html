---
layout: nil
---
{% assign some_bool = true %}

## Testing a simple wrap
[- Testing Include -] → {% wrap foo.html %}[- {{ yield }} -]{% endwrap %}

## Local var passing
[- Testing Include var_test -] → {% wrap foo.html some_var="var_test" %}[- {{ yield }} -]{% endwrap %}

## Filter testing
[- TESTING INCLUDE -] → {% wrap foo.html | upcase %}[- {% assign foo = 'bar' %}{{ yield }} -]{% endwrap %}
[- TESTING FILTERS -] → {% wrap foo.html | replace:"Include","Filters" | upcase | %}[- {{ yield }} -]{% endwrap %}

## Conditional wrap
'' → '{% wrap foo.html unless true %}[- {{ yield }} -]{% endwrap %}'
'' → '{% wrap foo.html unless some_bool %}[- {{ yield }} -]{% endwrap %}'
[- Testing Include -] → {% wrap foo.html if true %}[- {{ yield }} -]{% endwrap %}
[- Testing Include -] → {% wrap foo.html if some_bool %}[- {{ yield }} -]{% endwrap %}

## Plugin wraps
[- include from plugin -] → {% wrap awesome-sauce:some-include.html %}[- {{ yield }} -]{% endwrap %}
[- Yo Dawg, I heard you like includes. -] → {% wrap theme:greet.html greeting="Yo Dawg" %}[- {{ yield }} -]{% endwrap %}

## Wrap render
[- Testing Render -] → {% wrap_render test_render/_f.html %}[- {{ yield }} -]{% endwrap_render %}

## Wrap yield
{% content_for test %}Testing wrap yield{% endcontent_for %}
[- Testing wrap yield -] → {% wrap_yield test %}[- {{ yield }} -]{% endwrap_yield %}
