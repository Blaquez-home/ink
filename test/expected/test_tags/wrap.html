

## Testing a simple wrap
[- Testing Include -] → [- Testing Include -]

## Local var passing
[- Testing Include var_test -] → [- Testing Include var_test -]

## Conditional wrap
'' → ''
'' → ''
[- Testing Include -] → [- Testing Include -]
[- Testing Include -] → [- Testing Include -]

## Plugin wraps
[- include from plugin -] → [- include from plugin -]
[- Yo Dawg, I heard you like includes. -] → [- Yo Dawg, I heard you like includes. -]

## Wrap render
[- Testing Render -] → [- Testing Render -]

